### ISSUE TITLE
Add Insta-Flow Training support

### ISSUE THREAD

**Issue by isamu-isozaki** [huggingface/diffusers#5256]
**Is your feature request related to a problem? Please describe.**
For hacktoberfest, I wanted to add in the Rectified Flow training objective which is what the instaflow paper, which can generate images in one forward pass, used to finetune stable diffusion.

**Describe the solution you'd like**
I'll make a small pr and argument to examples/text_to_image based on code from [here](https://github.com/gnobitab/RectifiedFlow/blob/main/ImageGeneration/losses.py) to examples/text_to_image loss. I'm thinking of testing it out with mnist to see if it works.

**Additional context**
The links to the papers:
[instaflow](https://arxiv.org/abs/2309.06380)
[rectified flow](https://arxiv.org/abs/2209.03003)

**Comment by Pevernow** [comment]
+1

**Comment by isamu-isozaki** [comment]
I made a blog [here](https://huggingface.co/blog/Isamu136/insta-rectified-flow) for my notes on how the rectified flow works and what needs to be done for this PR. I was wrong in that this will probably need to be a community example in itself so might not finish within October(so I'll probably do some small PRs to win Hacktoberfest too)

**Comment by isamu-isozaki** [comment]
The preliminary TODO list I came up with is 
- [] Figure out how to map epsilon to velocity
- [] Make a script to generate the latent noise, images, and text to save to the dataset
- [] Make rectified flow/reflow script
- [] Make distillation script

**Comment by isamu-isozaki** [comment]
@patrickvonplaten Hi! Do you think this should be 3 separate PRs?

**Comment by DN6** [comment]
@isamu-isozaki Go for it 👍🏽 If you're adding to the examples folder I thinking you can make it a single PR.

**Comment by isamu-isozaki** [comment]
I finished making the dataset part. I am currently doing reflow and for log_validation, I think I will need a new scheduler for this which just adds the output of the unet /total timesteps to the latent.

**Comment by spacewalkingninja** [comment]
> I finished making the dataset part. I am currently doing reflow and for log_validation, I think I will need a new scheduler for this which just adds the output of the unet /total timesteps to the latent.

hey, hows it going, got a model now?

**Comment by isamu-isozaki** [comment]
@spacewalkingninja Ah I made a training colab for a lora version and training rectified flows. I did notice the loss goes down but haven't tested for that much longer training yet. Honestly, if anyone can run [this](https://colab.research.google.com/drive/13GmggtFLnVj55i2XnVp-E_-tLgv2ThSm?usp=sharing) for a while that'll be great. I will be running on my end too

**Comment by haofanwang** [comment]
It's really helpful, thanks.
